olga/analytics/models.py

```diff
@@ -149,3 +156,111 @@ def worlds_students_per_country_statistics(cls):  # pylint: disable=invalid-name
                 world_students_per_country[country] += count
 
         return dict(world_students_per_country.items())
+
+    @staticmethod
+    def get_student_amount_percentage(country_count_in_statistics, all_active_students_in_statistics):
+        # pylint: disable=invalid-name
+        """
+        Calculate student amount percentage based on total countries amount and particular county amount comparison.
+
+        If percentage is too small and doesn't show real numbers
+        it will be changed to '~0' (around zero value, but not totally).
+        """
+        students_amount_percentage = format(
+            country_count_in_statistics / all_active_students_in_statistics * 100, '.2f'
+        )
+
+        if students_amount_percentage == '0.00':
+            students_amount_percentage = '~0'
+
+        return students_amount_percentage
+
+    @staticmethod
+    def is_country_exists(country):
+        """
+        Check if value is not a null.
+        """
+        if country != 'null':
+            return True
+
+    @staticmethod
+    def append_country_data_to_list(country_list, country, count, student_amount_percentage=None):
+        """
+        Make data for particular visualization format. Example for `datamap` or tabular view.
+        """
+        country_data = [str(pycountry.countries.get(alpha_2=country).alpha_3), count]
+
+        if student_amount_percentage is not None:
+            country_data += [student_amount_percentage]
+
+        country_list.append(country_data)
+
+    @classmethod
+    def create_worlds_students_per_country_data_formatted_to_render(cls, worlds_students_per_country):
+        # pylint: disable=invalid-name
+        """
+        Create convenient and necessary data formats to render it from view.
```

Comment:
> I think we should have nice and convenient internal format and transform it for render on the last step. What you think?

It is a last step. View gets this data and render in context.



> Why list of tuples is convenient format?

It is list of lists now, after refactor.



> Which formats are convenient for processing/storage and which are for passing to backend?

Acceptor gets students per country as json dumped dictionary.
It is convenient for calculating `student without country`.
This format is `OK` for storage it in database.

Examples in `datamap` library say we are free to use `list of lists`. Why not?